jwst/datamodels/container.py

```diff
@@ -315,36 +280,27 @@ def from_asn(self, asn_data):
         try:
             for member in sublist:
                 filepath = op.join(asn_dir, member['expname'])
-                update_model = any(attr in member for attr in RECOGNIZED_MEMBER_FIELDS)
-                if update_model or self._save_open:
-                    m = datamodel_open(filepath, memmap=self._memmap)
-                    m.meta.asn.exptype = member['exptype']
-                    for attr, val in member.items():
-                        if attr in RECOGNIZED_MEMBER_FIELDS:
-                            if attr == 'tweakreg_catalog':
-                                if val.strip():
-                                    val = op.join(asn_dir, val)
-                                else:
-                                    val = None
-
-                            setattr(m.meta, attr, val)
-
-                    if not self._save_open:
-                        m.save(filepath, overwrite=True)
-                        m.close()
-                else:
-                    m = filepath
-
+                m = datamodel_open(filepath, memmap=self._memmap)
+                m.meta.asn.exptype = member['exptype']
+                for attr, val in member.items():
+                    if attr in RECOGNIZED_MEMBER_FIELDS:
+                        if attr == 'tweakreg_catalog':
+                            if val.strip():
+                                val = op.join(asn_dir, val)
+                            else:
+                                val = None
+
+                        setattr(m.meta, attr, val)
                 self._models.append(m)
 
         except IOError:
             self.close()
             raise
 
-        # Pull the whole association table into meta.asn_table
-        self.meta.asn_table = {}
+        # Pull the whole association table into the asn_table attribute
+        self.asn_table = {}
```

Comment:
```suggestion
        self.asn_table = copy.deepcopy(asn_data)
```
Would this work to remove the `properties.merge_tree` usage?

Reply:
this does appear to work, will update

Reply:
This is removed now, lmk how it looks @braingram